core/lib/instrument_api.h

```diff
@@ -1801,11 +1801,63 @@ DR_API
 bool
 dr_set_client_version_string(const char *version);
 
+DR_API
+/**
+ * Returns the error code of the last failed API routine. Users should check whether or
+ * not the API routine that they just called has failed prior to this function.
+ *
+ * \warning Not all API routines currently support the registering of an error code upon
+ * their failure. Therefore, check the routine's documentation to see whether it supports
+ * setting error codes.
+ */
+dr_error_code_t
+dr_get_error_code(void);
+
 DR_API
 /** Returns the image name (without path) of the current application. */
 const char *
 dr_get_application_name(void);
 
+DR_API
+/**
+ * Provides information about the app's arguments by setting \p args_buf up to
+ * the size \p buf_size. Returns the number of args set or -1 on error.
```

Comment:
I would mention that dr_num_app_args() can be used to query the total number of args available.

Reply:
Done.